packages/astro/src/core/routing/match.ts

```diff
@@ -2,7 +2,12 @@ import type { ManifestData, RouteData } from '../../@types/astro.js';
 
 /** Find matching route from pathname */
 export function matchRoute(pathname: string, manifest: ManifestData): RouteData | undefined {
-	return manifest.routes.find((route) => route.pattern.test(decodeURI(pathname)));
+	return manifest.routes.find((route) => {
+		return (
+			route.pattern.test(decodeURI(pathname)) ||
+			route.fallbackRoutes.some((fallbackRoute) => fallbackRoute.pattern.test(decodeURI(pathname)))
+		);
+	});
```

Comment:
While we're updating this, it would be nice if the `decodeURI(pathname)` part is extracted as a variable for a little more perf.